### PR TITLE
[3.12] gh-115664: Fix chronological ordering of versionadded and versionchanged directives (GH-115676)

### PR DESCRIPTION
**Pull request by brianschubert**
(cherry picked from commit b02ab65e8083185b76a7dd06f470b779580a3b03)

* Issue: gh-115664

----
📚 Documentation preview 📚: https://cpython-previews--115678.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/sys.rst [hunk 1]
```diff
@@ -16,11 +16,13 @@ always available.
...
    script, this contains the ABI flags as specified by :pep:`3149`.
 
+   .. versionadded:: 3.2
+
    .. versionchanged:: 3.8
       Default flags became an empty string (``m`` flag for pymalloc has been
       removed).
 
-   .. versionadded:: 3.2
+   .. availability:: Unix.
```
Comment (AlexWaygood) [inline Doc/library/sys.rst:25]: N.B. The addition of this availability note wasn't part of the original PR; it looks like it existed on the `main` branch but not the `3.12` branch (and this was probably (at least part of) the cause of the merge conflict). _Strictly_ speaking, therefore, this change doesn't belong in this PR.

However, I'm fine with adding it to the 3.12 branch here; it's true on Python 3.12 as well as Python 3.13.

**Review by AlexWaygood — APPROVED** [review]
Thanks!

File: Doc/library/sys.rst (hunk 1 above)
Comment (brianschubert) [reply Doc/library/sys.rst:25]: D'oh! I noticed this when doing the 3.11 backport. I came back here to fix it, only to find that it had already been merged!

Comment (AlexWaygood) [reply Doc/library/sys.rst:25]: No worries!

**Pull request merged**